src/plots/interfaces.ts

```diff
@@ -1,55 +1,179 @@
 
 import { Tuplet, Triplet, Domain } from '../common/interfaces';
 
+/**
+ * Data format used by plots
+ * @example [[0, 10], [10, 21], [20, 19] ... ]
+ */
 export type PlotData = Tuplet<number>[];
 
+/**
+ * Data format used by differential plot
+ * @example [[0, 10. 0.5], [10, 21, 0.78], [20, 19, 1.21] ... ]
+ */
 export type DifferentialPlotData = Triplet<number>[];
 
+/**
+ * Function to control what data to plot, where
+ * y is the value/range part of a plot data tuplet
+ * and x is the domain part of a prop data tuplet
+ * @example (y, x) => Number.isFinite(y) && Number.isFinite(x)
+ */
 export type DefinedFunction = (y?:number, x?:number) => boolean;
 
+/**
+ * When used in combination with a Track, it is the track's responsibillity to fetch
+ * and pass down data to the plots. This is configured by providing a data accessor
+ * callback function, where the track's data are made available as the input argument
+ * and the returned value will be used by the plot.
+ * @example data => data //default
+ * @example data => data.dataSeries[1].dataPoints
+ */
 export type DataAccessorFunction = (data: any) => (PlotData|[PlotData, PlotData]|any);
 
+/**
+ * Base interface for plot options
+ */
 export interface PlotOptions {
+  /**
+   * offset position to draw the plot
+   * @example 0.5 // renders the plot 50 % off from origin
+   */
   offset?: number,
+  /**
+   * Scale type: 'linear' or 'log'
+   */
   scale?: string,
+  /**
+   * Y axis (range) for plot data
```

Comment:
is the domain still only for y axis? could use example/default values

Reply:
Will fix in #24